react/react1/week2/todo-tracker/src/App.js

```diff
@@ -0,0 +1,79 @@
+import "./App.css";
+import React from "react";
+import ToDoListComp from "./components/ToDoList";
+import toDoList from "./data/toDoList";
+import SearchComp from "./components/Search";
+import AddToDo from "./components/AddToDo";
+import WatchCountComp from "./components/WatchCount";
+
+function App() {
+  //here two useState(toDoList) are set because once the search is set the
+  // fulltodo list gets modified and shorten. Then the other functionalities will be
+  // effective on the shorten list once the search bar is cleared. So, the second
+  //useState(toDoList) is used to get the full todolist once the search bar is cleared.
+  const [filterToDoList, setFilterToDoList] = React.useState(toDoList);
+  const [fullToDoList, setFullToDoList] = React.useState(toDoList);
+
+  return (
+    <div className="App">
+      <WatchCountComp />
+
+      <SearchComp
+        onFilteredToDoListChange={(text) => {
```

Comment:
Would suggest defining complex functions like this outside of your JSX template or the template can get a big bloated.